proof/refine/ARM/StateRelation.thy

```diff
@@ -882,7 +882,7 @@ lemma sc_replies_relation_scReply:
   "\<lbrakk> sc_replies_relation s s';
      kheap s x = Some (kernel_object.SchedContext sc n);
      ksPSpace s' x = Some (KOSchedContext sc')\<rbrakk>
-    \<Longrightarrow> hd_opt (sc_replies sc) = scReply sc'"
+    \<Longrightarrow> hd_opt (sc_replies sc) = scReplies_of s' x"
```

Comment:
Hmm. I think I told you something not quite true when you asked me about this yesterday. We can do a projection version, actually.
```
"⟦ sc_replies_relation s s'; sc_at x s; bound (scs_of' s' x)⟧
    ⟹ (sc_replies_of s |> hd_opt) x = scReplies_of s' x"
```
I believe this would work. Can you have a look into this?

Reply:
Also, we probably want to rename the lemma with something more suitable.